skimage/filters/rank/generic.py

```diff
@@ -203,24 +290,25 @@ def autolevel(image, selem, out=None, mask=None, shift_x=False, shift_y=False):
                                    shift_x=shift_x, shift_y=shift_y)
 
 
-def bottomhat(image, selem, out=None, mask=None, shift_x=False, shift_y=False):
+def bottomhat(image, selem=None, out=None, mask=None, shift_x=False,
+              shift_y=False):
     """Local bottom-hat of an image.
 
     This filter computes the morphological closing of the image and then
     subtracts the result from the original image.
 
     Parameters
     ----------
-    image : 2-D array (uint8, uint16)
+    image : 2-D array (integer, float or boolean)
         Input image.
-    selem : 2-D array
+    selem : 2-D array (integer, float, boolean or optional)
```

Comment:
Bottom hat doesn't seem to specify default `selem`. Did you want to add a default to all rank filters? if so, if might be appropriate to put it in the `_handle_input` function.

Reply:
I may have overdone it trying to address feedback above to add `optional` to the docstrings for `selem`. I'm not sure what's the best policy for this.

Reply:
@hmaarrfk What is the best policy to add the defaults? I'm not sure what makes the most sense in this context.

Reply:
Generally it comes in an as needed basis. People make requests as to what makes sense. It is often hard to impose a "most useful" default and one that makes the "most untiuitive sense"

Reply:
Ok, I'm going to try adding them into `_handle_input` then.